cocos/serialization/compiled/typed-array.ts

```diff
@@ -0,0 +1,125 @@
+import { assertIsTrue, sys } from '../../core';
+import { IRuntimeFileData } from '../deserialize';
+
+assertIsTrue(sys.isLittleEndian, `Deserialization system currently suppose little endian.`);
+
+export const typedArrayTypeTable = Object.freeze([
+    Float32Array,
+    Float64Array,
+
+    Int8Array,
+    Int16Array,
+    Int32Array,
+
+    Uint8Array,
+    Uint16Array,
+    Uint32Array,
+
+    Uint8ClampedArray,
+    // BigInt64Array,
+    // BigUint64Array,
+] as const);
+
+/**
+ * Describes the serialized data of an typed array.
+ * - If it's an array, it's `TypedArrayDataJson`.
+ * - Otherwise, it's `TypedArrayDataPtr`.
+ */
+export type TypedArrayData = TypedArrayDataJson | TypedArrayDataPtr;
+
+export type TypedArrayDataJson = [
+    /**
+     * Indicates the constructor of typed array.
+     * It's index of the constructor in `TypedArrays`.
+     */
+    typeIndex: number,
+
+    /**
+     * Array element values.
+     */
+    elements: number[],
+];
+
+/**
+ * Let `offset` be this value,
+ * Let `storage` be the binary buffer attached to the deserialized document.
+ * Then, the data of `storage` started from `offset`
+ * can be described using the following structure(in C++, assuming fields are packed tightly):
+ *
+ * ```cpp
+ * struct _ {
+ *   /// Indicates the constructor of typed array.
+ *   /// It's index of the constructor in `typedArrayTypeTable`.
+ *   std::uint32_t typeIndex;
+ *
+ *   /// The typed array's element count. Note this is not "byte length".
+ *   std:: uint32_t length;
+ *
+ *   /// Automatically padding bytes to align the `arrayBufferBytes`.
+ *   /// See comments on `arrayBufferBytes`.
+ *   std::byte[] padding;
```

Comment:
padding -> _padding

Reply:
Solved in a31fbee938236cc9af8326d454119bd7c2864c13